packages/back-end/src/api/data-sources/listDataSources.ts

```diff
@@ -0,0 +1,33 @@
+import { z } from "zod";
+import { ListDataSourcesResponse } from "../../../types/openapi";
+import {
+  getDataSourcesByOrganization,
+  toDataSourceApiInterface,
+} from "../../models/DataSourceModel";
+import { applyPagination, createApiRequestHandler } from "../../util/handler";
+
+export const listDataSources = createApiRequestHandler({
+  querySchema: z
+    .object({
+      limit: z.string().optional(),
+      offset: z.string().optional(),
+    })
+    .strict(),
+})(
+  async (req): Promise<ListDataSourcesResponse> => {
+    const dataSources = await getDataSourcesByOrganization(req.organization.id);
+
+    // TODO: Move sorting/limiting to the database query for better performance
```

Comment:
Not sure if this is moved or new, but do we have a link to a ticket either in Linear or Github issues so this gets addressed?